client/landing/stepper/declarative-flow/internals/steps-repository/sensei-purpose/index.tsx

```diff
@@ -19,25 +21,63 @@ import './style.scss';
 
 const wait = ( ms: number ) => new Promise( ( res ) => setTimeout( res, ms ) );
 
+type ResponseError = {
+	error: string;
+};
+
+const setCourseTheme = async ( site: number, variation: string | null = null ) =>
+	wpcomRequest( {
+		path: `/sites/${ site }/themes/mine`,
+		method: 'POST',
+		apiVersion: '1.1',
+		body: {
+			theme: 'course',
+			...( variation && { style_variation_slug: variation } ),
+		},
+	} );
+
+const installCourseTheme = async ( siteId: number ) =>
+	wpcom.req.post( { path: `/sites/${ siteId }/themes/course/install` } );
+
 const SenseiPurpose: Step = ( { navigation: { submit } } ) => {
 	const [ progress, setProgress ] = useState< number >( 0 );
+	const urlParams = new URLSearchParams( window.location.search );
+	const siteId = parseInt( urlParams.get( 'siteId' ) ?? '0' );
+	const selectedVariation = urlParams.get( 'variation' ) ?? 'green';
 
 	useEffect( () => {
 		clearSelectedPurposes();
 	}, [] );
 
 	// Stall for a few seconds while the atomic transfer is going on.
 	useEffect( () => {
-		const tick = async () => {
-			if ( progress < 110 ) {
-				await wait( 1500 );
-				setProgress( ( progress ) => progress + 33 );
+		const installTheme = async () => {
```

Comment:
I'm going to try to move this into the steps repository instead. Maybe I can remove most of the component's code since all it does is wait for the site to be transferred to Atomic which I already handle with the install theme logic, and then show the purposes screen.

Reply:
Moved.